samples/net/wifi/boards/m5stickc_plus.conf

```
@@ -0,0 +1,11 @@
+CONFIG_WIFI=y
+
+CONFIG_NETWORKING=y
+CONFIG_NET_L2_ETHERNET=y
+
+CONFIG_NET_IPV6=n
+CONFIG_NET_IPV4=y
+CONFIG_NET_DHCPV4=y
+CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4=y
+
+CONFIG_NET_LOG=y
```

I realize this is a copy paste from other overlays, but it looks like several options are already set at the prj.conf level.
It looks like only the below would be needed? @sylvioalves is that correct, and would it make sense to clean up other ESP32 overlays?

```suggestion
CONFIG_NET_L2_ETHERNET=y
CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4=y
```

---

@kartben You are correct. Eventually all ESP32 boards could enable by default `CONFIG_NET_L2_ETHERNET` as it is the only way right now. Finally, `CONFIG_ESP32_WIFI_STA_AUTO_DHCPV4` is there so that DHCP is automatically started and make sure it gets IPv4 address. Can this clean up be merge within feature freeze window?

---

> Can this clean up be merged within feature freeze window?

I think this would be left at the discretion of the release engineers as it could be considered trivial if it's literally just removing duplicate entries.
I wonder if the cleanup couldn't be more something along the lines of trying to provide a generic ESP32 overlay? It would be less "magic" as people would need to provide the path to it when building, but oth it would work for all ESP32 boards, since right now it doesn't seem to scale all that well to have as many overlay files (x2) as there are ESP32 boards?